src/mp3.c

```diff
@@ -0,0 +1,354 @@
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU Lesser General Public License for more details.
+**
+** You should have received a copy of the GNU Lesser General Public License
+** along with this program; if not, write to the Free Software
+** Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+*/
+
+#include	"sfconfig.h"
+
+#include	<stdio.h>
+#include	<fcntl.h>
+#include	<string.h>
+#include	<dlfcn.h>
+#include	<ctype.h>
+#include	<mpg123.h>
+#include	<lame/lame.h>
+
+#include	"sndfile.h"
+#include	"sfendian.h"
+#include	"common.h"
+
+/*------------------------------------------------------------------------------
+** Macros to handle big/little endian issues.
+*/
+
+/*------------------------------------------------------------------------------
+** Typedefs.
+*/
+
+typedef struct
+{
+	lame_global_flags * gfp ;
+	unsigned char * encbuffer ;
+} MP3_WRITE_PRIVATE ;
+
+typedef struct
+{	unsigned		n_instances ;
+	void			*dlhandle ;
+	lame_global_flags *	(*init) (void) ;
+	void			(*close) (lame_global_flags *) ;
+	int			(*init_params) (lame_global_flags *) ;
+	int			(*encode_interleaved) (
+		lame_global_flags *, short *, int, unsigned char *, int) ;
+} LAME_LIB_HANDLE ;
+
+/*------------------------------------------------------------------------------
+** Private static functions.
+*/
+static	int		mp3_read_open		(SF_PRIVATE * psf) ;
+static	int		mp3_read_close		(SF_PRIVATE * psf) ;
+static	int		mp3_read_header		(SF_PRIVATE * psf, mpg123_handle * decoder) ;
+static	sf_count_t	mp3_read_2s		(SF_PRIVATE * psf, short * ptr, sf_count_t len) ;
+static	sf_count_t	mp3_read_2i		(SF_PRIVATE * psf, int * ptr, sf_count_t len) ;
+static	sf_count_t	mp3_read_2f		(SF_PRIVATE * psf, float * ptr, sf_count_t len) ;
+static	sf_count_t	mp3_read_2d		(SF_PRIVATE * psf, double * ptr, sf_count_t len) ;
+static	sf_count_t	mp3_read_seek		(SF_PRIVATE *psf, int whence, sf_count_t offset) ;
+static	ssize_t		mp3_read_sf_handle	(void * handle, void * buffer, size_t bytes) ;
+static	off_t		mp3_seek_sf_handle	(void * handle, off_t offset, int whence) ;
+static	int		mp3_format_to_encoding	(int encoding) ;
+
+static	char *		mp3_init_lame		(void) ;
+static	void		mp3_close_lame		(void) ;
+
+static	int		mp3_write_open		(SF_PRIVATE * psf) ;
+static	int		mp3_write_close		(SF_PRIVATE * psf) ;
+static	sf_count_t	mp3_write_2s		(SF_PRIVATE *psf, const short *ptr, sf_count_t len) ;
+
+// FIXME: This initialisation should have a better hook
+static int mpg123_initialised = 0 ;
+
+/*------------------------------------------------------------------------------
+** Public function.
+*/
+
+int
+mp3_open (SF_PRIVATE * psf)
+{	psf->codec_data = NULL ;
+	switch (psf->file.mode) {
+	case (SFM_READ) :
+		return mp3_read_open (psf) ;
+	case (SFM_WRITE) :
+		return mp3_write_open (psf) ;
+	default:
+		return SFE_UNIMPLEMENTED ;
+	}
```

Comment:
The libsndfile codebase has an coding style which it adheres to with a reasonably high level of consistency.

The is an a Python script `Scripts/cstyle.py` which catchs a large number of issues. You should still have a look at the other code and match the style.

Reply:
I thought I had got it all past cstyle.py (which I believe is triggered in the pre-commit hook), but I'll check elsewhere for the preferred bracket style.